File: MES_EQUIPMENT/Models/MoldingInspectionDto.cs
public class MoldingInspectionDto
{
    public int SampleSize { get; set; }                  // 샘플링 수량
    public string? InspectionCriteria { get; set; }      // 검사 기준
    public double ThicknessPassRatio { get; set; }       // 두께 합격률
    public double VoidPassRatio { get; set; }            // 보이드 합격률
    public double CrackPassRatio { get; set; }           // 크랙 합격률
    public double OverallPassRatio { get; set; }         // 전체 합격률

    // ✅ 역직렬화: byte[] → 검사 DTO
    public static MoldingInspectionDto FromBytes(byte[] payload)
    {
        int offset = 0;
        MoldingInspectionDto dto = new MoldingInspectionDto();

        dto.SampleSize = BitConverter.ToInt32(payload, offset); offset += 4;

        int strLen = BitConverter.ToInt16(payload, offset); offset += 2;
        if (strLen > 0)
        {
            dto.InspectionCriteria = System.Text.Encoding.UTF8.GetString(payload, offset, strLen);
            offset += strLen;
        }

        dto.ThicknessPassRatio = BitConverter.ToDouble(payload, offset); offset += 8;
        dto.VoidPassRatio = BitConverter.ToDouble(payload, offset); offset += 8;
        dto.CrackPassRatio = BitConverter.ToDouble(payload, offset); offset += 8;
        dto.OverallPassRatio = BitConverter.ToDouble(payload, offset); offset += 8;

        return dto;
    }
}

File: MES_SPRING/src/main/java/com/hm/mes_final_260106/dto/MoldingInspectionDto.java
package com.hm.mes_final_260106.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MoldingInspectionDto {

    private Long id;                    // PK
    private Long moldingId;             // FK: Molding 공정 참조
    private int sampleSize;             // 샘플링 수량
    private String inspectionCriteria;  // 검사 기준

    private double thicknessPassRatio;  // 두께 합격률 (%)
    private double voidPassRatio;       // 보이드(기포) 합격률 (%)
    private double crackPassRatio;      // 크랙(균열) 합격률 (%)
    private double overallPassRatio;    // 전체 합격률 (%)
}
